doc/src/manual/profile.md

```diff
@@ -338,15 +338,78 @@ argument can be passed to speed it up by making it skip some allocations.
 Passing `sample_rate=1.0` will make it record everything (which is slow);
 `sample_rate=0.1` will record only 10% of the allocations (faster), etc.
 
-!!! note
+!!! compat "Julia 1.11"
+
+    Older versions of Julia could not capture types in all cases. In older versions of
+    Julia, if you see an allocation of type `Profile.Allocs.UnknownType`, it means that
+    the profiler doesn't know what type of object was allocated. This mainly happened when
+    the allocation was coming from generated code produced by the compiler. See
+    [issue #43688](https://github.com/JuliaLang/julia/issues/43688) for more info.
+
+    Since Julia 1.11, all allocations should have a type reported.
+
+For more details on how to use this tool, please see the following talk from JuliaCon 2022:
+https://www.youtube.com/watch?v=BFvpwC8hEWQ
+
+##### Allocation Profiler Example
+
+Here is an example of how to invoke the Allocation profiler. A good number of samples to aim
+for is around 1 - 10 thousand. Too many, and the profile visualizer can get overwhelmed, and
+profiling will be slow. Too few, and you don't have a representative sample.
+
+```julia-repl
+julia> import Profile
+
+julia> @time my_function()  # Estimate allocations from a (second-run) of the function
+  0.110018 seconds (1.50 M allocations: 58.725 MiB, 17.17% gc time)
+500000
+
+julia> Profile.Allocs.clear()
+
+julia> Profile.Allocs.@profile sample_rate=0.001 begin   # 1.5 M * 0.001 = ~1.5K allocs.
+           my_function()
+       end
+500000
+
+julia> prof = Profile.Allocs.fetch();
```

Comment:
This is not required, right?

Most people will just copy the example. So in my view the example should be the simplest thing possible.

Something like:

```
using Profile, PProf
Profile.Allocs.clear()
Profile.Allocs.@profile sample_rate=0.001 my_function()
PProf.Allocs.pprof()
```

Reply:
Thanks for the review. Thanks that was excellent feedback! :)

I've added this very simple example to the doc as the first example, and then i explain more about why i was going into the details in the second one. 👍 Let me know if that looks good? (We can also do follow-up PRs if you want to add more!